Docs/Packaging guide: Add BundlePackage (#39691)
LLNL/spack: lib/spack/docs/packaging_guide.rst

       version("8.2.0", md5="1c9f62f0778697a09d36121ead88e08e")
       version("8.1.2", md5="d47dd09ed7ae6e7fd6f9a816d7f5fdf6")

.. note::

>    Versions should be listed in descending order, from newest to oldest.

Reviewer: ```suggestion
   By convention, we list versions in descending order, from newest to oldest.
```
